src/satellite/process.ts

```diff
@@ -454,13 +569,126 @@ export class SatelliteProcess implements Satellite {
     return rows as unknown as OplogEntry[]
   }
 
+  async _getUpdatedEntries(
+    timestamp: Date,
+    since?: number
+  ): Promise<OplogEntry[]> {
+    if (since === undefined) {
+      since = this._lastAckdRowId
+    }
+    const oplog = this.opts.oplogTable.toString()
+
+    const selectChanges = `
+      SELECT * FROM ${oplog}
+      WHERE timestamp = ? AND
+            rowid > ?
+      ORDER BY rowid ASC
+    `
+
+    const rows = await this.adapter.query({
+      sql: selectChanges,
+      args: [timestamp.toISOString(), since],
+    })
+    return rows as unknown as OplogEntry[]
+  }
+
+  async _getOplogShadowEntry(oplog: OplogEntry): Promise<ShadowEntry[]> {
+    const shadow = this.opts.shadowTable.toString()
+    let query
+    let selectTags = `SELECT * FROM ${shadow}`
+    if (oplog != undefined) {
```

Comment:
If `oplog` can be undefined, let's reflect that in the types of the argument (here and up the call chain). Otherwise, let's drop the condition